tools/sof-kernel-log-check.sh

```diff
@@ -3,9 +3,23 @@
 begin_line=${1:-1}
 declare err_str ignore_str project_key
 err_str="error|failed|timed out|panic|oops"
-ignore_str="error: debugfs write failed to idle -16|error: status|iteration [01]"
+ignore_str=""
 project_key="sof-audio"
 
+# There will be debug logs at each failed initializaiton of DSP:
+#   sof-audio-pci 0000:00:1f.3: error: status = 0x00000000 panic = 0x00000000
+#   sof-audio-pci 0000:00:1f.3: error: Error code=0xffffffff: FW status=0xffffffff
+#   sof-audio-pci 0000:00:1f.3: error: iteration 0 of Core En/ROM load failed: -110
+# We will reinit DSP if it is failed to init, and retry 3 times, so the errors in
+# debug logs at the frist and second retry can be ignored.
+# Check https://github.com/thesofproject/linux/pull/1676 for more information.
+# TODO explain for "error: Error ..."
+ignore_str="$ignore_str"'|error: iteration [01]'
```

Comment:
Wait, I just realized the string starts with a vertical bar. That means `ignore_str`probably matches anything.

How did you test this?

Reply:
@marc-hb the ignored item here is not 100% reproduced, and some may need a lot of stress test. No good way to test but just run script without error. 

Maybe initialize ignore_str with the first ignore item? This will make code style not unified.

Reply:
Is this OK for you?
```
ignore_str='error: iteration [01]'
ignore_str="$ignore_str"'|error: status'
```

Reply:
This version looks  better but do test the code change locally with a couple errors that are easy to reproduce: at lest one ignored error and another  not ignored. Then change the ignored message(s) and only the message(s) just before submitting.

This part of the code is very critical because a single, one character bug like this vertical bar can turn the entire results green. So it must be very thoroughly tested.